### PR TITLE
boards/doc: Add a guide to find physical board pins

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

This is a guide to locating the physical location of pins on a board.  If I want to connect an i2c device to a samr21-xpro and I don't already have experience this should be a good place to start when trying to figure out which pins to plug.

### Testing procedure

Make doc and read the output.  Check spelling and grammar (mine is known to be bad).  Make sure it makes sense

### CODE REVIEW

**Comment by jcarrano** [comment]
Kind of off-topic, but one thing I discovered not long ago (shame on me) is that when writing the TXTs meant to go into dogygen, it is not necessary to have a left column of "*" characters.

At least in my editor those "*" make word wrapping quite harder.

**Comment by MrKevinWeiss** [comment]
Thank you, I thought it was so silly to do that.

**Comment by jcarrano** [comment]
I guess it is a case of _cargo-cult_-ism (I also fell victim to it, so I won't complain much)

**Comment by MrKevinWeiss** [comment]
@jcarrano done!

**Comment by MrKevinWeiss** [comment]
Is there any tags or something I can add so that when I refer to it it will jump to that section.  Something like the #c_pinn in https://github.com/riot-appstore/PHiLIP#c_pinn  ?

**Comment by MrKevinWeiss** [comment]
@smlng ping

File: boards/doc.txt [hunk 1]
```diff
@@ -15,4 +15,52 @@
...
+## Guide to board pinouts
+
+The following are the steps to take to locate specific pinouts on boards.
+This is useful when connecting external hardware to RIOT supported boards.
+
+1. Check the board documentation to see if the pinout is available
+2. If the pinout is not available or up to date check the following:
+    - `boards/<TARGET_BOARD>/include/periph_conf.h` for board specific
+peripherals such as uart  or spi pins
+    - `boards/<TARGET_BOARD>/include/board.h` for GPIO specific pins such
```
Comment (smlng) [inline boards/doc.txt]: For variable part I would stick with already used (environment) variables, i.e. we never user `TARGET_BOARD` its either simply `BOARD` or `RIOT_BOARD` - don't introduce new names that will only confuse

File: boards/doc.txt [hunk 2]
```diff
@@ -15,4 +15,52 @@
...
+as `LED0` or `BTN0`
+    - `boards/common/<COMMON_BOARD_OR_CPU>/include/periph_conf_common.h`
```
Comment (smlng) [inline boards/doc.txt]: `_OR_CPU`??

File: boards/doc.txt [hunk 3]
```diff
@@ -15,4 +15,52 @@
...
+pb8 and pb9
+3. Once the needed pins are determined with either a port or number and pin
+number, such as `PA, 1` or `0, 1` correlate that to the board (the board
+may either have the designators or search online to find which gpio
+corresponds to physical location on the board)
+4. Match the `GPIO` with the specific pin and connect the wires
+
+### Examples of finding pinouts
+The following will be different examples of how to find pinouts
+#### Finding pins of the i2c device 0 on samr21-xpro
```
Comment (smlng) [inline boards/doc.txt]: add empty line above

**Comment by jcarrano** [comment]
@MrKevinWeiss you mean something like this?

https://github.com/RIOT-OS/RIOT/blob/395aaf34e9c63aa760cfc5e4371468c561c37c11/doc/doxygen/src/advanced-build-system-tricks.md#L6-L7

**Comment by smlng** [comment]
compile doc looks good btw.

**Comment by MrKevinWeiss** [comment]
Thanks

**Comment by MrKevinWeiss** [comment]
I added the tag, thanks!  smlng have all your comments been addressed?

**Comment by smlng** [comment]
> I added the tag, thanks! smlng have all your comments been addressed?

I don't see them addressed, did you push everything?

**Comment by MrKevinWeiss** [comment]
I actually can see where your blocking aside from "compile doc looks good btw." but it says you have requested changes...

**Comment by MrKevinWeiss** [comment]
hmmm I guess refresh helps :laughing:

File: boards/doc.txt (hunk 1 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: Yup

File: boards/doc.txt (hunk 2 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: I mean it isn't always a board or board series in boards/common.  See `nucleo`,  `nucleo-64`, `stm32`, `stm32f103c8` etc.

It is pretty hard to explain that in one word.

File: boards/doc.txt (hunk 3 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: yup

File: boards/doc.txt (hunk 2 above)
Comment (smlng) [reply boards/doc.txt]: yeah right, the naming is (once again in RIOT) not consistent and doesn't follow a good/common scheme - well leave as is for now

File: boards/doc.txt [hunk 4]
```diff
@@ -15,4 +15,52 @@
...
+as `LED0` or `BTN0`
+    - `boards/common/<COMMON_BOARD_OR_CPU>/include/periph_conf_common.h`
+for shared pins that are the same for many different boards (board_common.h
+or others will be available too)
+    - `boards/common/<COMMON_BOARD_OR_CPU>/include/<PERIPH_DECLARATION>`
+also may contain specific periphirals such as
+`PERIPH_DECLARATION = cfg_i2c1_pb8_pb9.h` which indicates i2c pins are on
+pb8 and pb9
+3. Once the needed pins are determined with either a port or number and pin
+number, such as `PA, 1` or `0, 1` correlate that to the board (the board
```
[the remark this answers is not in the document]
Comment (smlng) [reply boards/doc.txt]: this is just to make sure (new) users don't get confused, bc as below: the naming and usage of `PORTX`, `PORT_X` or `PX` is not consistent

**Comment by MrKevinWeiss** [comment]
I guess I shouldn't always force push...

File: boards/doc.txt (hunk 2 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: yup

**Comment by MrKevinWeiss** [comment]
@smlng ping when you have time.

File: boards/doc.txt [hunk 5]
```diff
@@ -15,4 +15,56 @@
...
+Guide to board pinouts {#pinout_guide}
+=====================================
+
+The following are the steps to take to locate specific pinouts on boards.
+This is useful when connecting external hardware to RIOT supported boards.
+
+1. Check the board documentation to see if the pinout is available
+2. If the pinout is not available or up to date check the following:
+    - `boards/<BOARD>/include/periph_conf.h` for board specific
+peripherals such as uart  or spi pins
```
Comment (smlng) [inline boards/doc.txt]: I'm always unsure: capitalise abbreviations? i.e. UART and SPI?

Also there are 2 spaces after uart, one should be enough

File: boards/doc.txt [hunk 6]
```diff
@@ -15,4 +15,56 @@
...
  * device drivers should go into this module.
+ *
+
+Guide to board pinouts {#pinout_guide}
+=====================================
+
+The following are the steps to take to locate specific pinouts on boards.
+This is useful when connecting external hardware to RIOT supported boards.
+
+1. Check the board documentation to see if the pinout is available
```
Comment (smlng) [inline boards/doc.txt]: where to find that (as a newbie)? I.e. add a link to http://doc.riot-os.org/group__boards.html

File: boards/doc.txt [hunk 7]
```diff
@@ -15,4 +15,56 @@
...
+for shared pins that are the same for many different boards (board_common.h
+or others will be available too)
+    - `boards/common/<COMMON_BOARD_OR_CPU>/include/<PERIPH_DECLARATION>`
+also may contain specific periphirals such as
+`PERIPH_DECLARATION = cfg_i2c1_pb8_pb9.h` which indicates i2c pins are on
+pb8 and pb9
+3. Once the needed pins are determined with either a port where
+`PORTA = PORT_A = PA = 0` or `PORTB = PORT_B = PB = 1` and pin
+number, such as `PA, 1`, correlate that to the board (the board
+may either have the designators or search online to find which gpio
```
Comment (smlng) [inline boards/doc.txt]: /gpio/GPIO/ ?

File: boards/doc.txt (hunk 5 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: Thanks, done.

File: boards/doc.txt (hunk 6 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: Done

File: boards/doc.txt (hunk 7 above)
Comment (MrKevinWeiss) [reply boards/doc.txt]: Done, went with all caps.

**Comment by MrKevinWeiss** [comment]
@smlng, have your comments been addressed?

**Comment by smlng** [comment]
please squash

**Comment by smlng** [comment]
and trigger ci then

**Pull request merged**